### PR TITLE
weekend 2 assesment

### PR DESCRIPTION
**Pull request by RichDTran**
# Submitting Week 2 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [ ] Coding Challenges
- [ ] Coding Concepts
- [ ] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -13,47 +13,152 @@
...
+// const divisible3 = (object) => {
+//     if(object.number % 3 === 0){
+//         return `${object.number} is divisible by three`
+//     }   else {
+//         return `${object.number} is not divisible by three`
+//     }
+// }
+
+// describe("divisible3", () => {
+//     it("returns a statement with the appropriate coffee order", () => {
```
Comment (Genemichael25) [inline code-challenges.test.js:66]: The it block should reflect what you want the test to check for. Coffee order for this challenge question doesn't seem appropriate.

File: code-challenges.test.js [hunk 2]
```diff
@@ -13,47 +13,152 @@
...
 
-// b) Create the function that makes the test pass.
+// // b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in a string and logs the index of the first vowel.
+// const arrayCap = (array) => {
+//     return array.map(value => {
+//         return value.charAt().toUpperCase() +value.slice(1)
+//     })
+// }
```
Comment (Genemichael25) [inline code-challenges.test.js:95]: Good job with this function! Make sure you include charAt and how you intend to use it in your pseudocode. You can also talk about how you will slice from index 1 and why that makes sense for this problem.

File: code-challenges.test.js [hunk 3]
```diff
@@ -13,47 +13,152 @@
...
+
+// // b) Create the function that makes the test pass.
+const vowelsA = ["a", "e", "i", "o", "u"]
+const indexVowel = array => {
+//     return value.length (value => {
+//         return value.indexOf() .includes(vowelsA => {
+//             return array})
+//         })
+// }
+
```
Comment (Genemichael25) [inline code-challenges.test.js:135]: I like that you are attempting different ways. This is an excellent example of where pseudocode will really help you think each step and understand if you are missing something. Things like understanding what .include will produce or if .length will be useful.

File: coding-concepts.js [hunk 4]
```diff
@@ -8,36 +8,45 @@
 // --------------------1) What will this log?
 
 const cohort = "Alpha 2023"
-// console.log(cohort.split(""))
+//console.log(cohort.split(""))
+
+// a) Your answer: It will space out each value such as "A" "l" "p" "h" "a" "2" "0" "2" "3"
+// b) Verify and explain: My answer was correct besides the "" , .split seperates each value
+
```
Comment (Genemichael25) [inline coding-concepts.js:15]: Great job! Keep in mind that the space will also be included in the array

File: coding-concepts.js [hunk 5]
```diff
@@ -8,36 +8,45 @@
...
 const onlyOdds = [11, 12, 13, 14, 15].filter((number) => number % 2 !== 0)
-// console.log(onlyOdds)
+ //console.log(onlyOdds)
 
-// a) Your answer:
-// b) Verify and explain:
+// a) Your answer: This function will go through each value and will divide by 2 and if 
+//   evenly divided and if there is a remainder will return through the log which would be "odd"
+// b) Verify and explain: [ 11, 13, 15 ] these values do not evenly divded by 2 and have a remainder
+// verifying these numbers are odd and are returned.
```
Comment (Genemichael25) [inline coding-concepts.js:49]: Exactly, great job explaining filter in this situation!

File: interview-questions.md [hunk 6]
```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+such as (array[bean, fruit, meat], /string"Hello How are you", index[2], 25) 
+where a argument an argument is if a value is given under a condition and then 
+it is logged in result as a return
+
+Researched answer: Parameters are the names listed in the functions Definition
+                    Example : function example(paramenter) {
+                        console.log(parameter)
+                    }
+                    An Argument are real values passed to the function
+                    const argument = "LEARN"
```
Comment (Genemichael25) [inline interview-questions.md:21]: We can think of a parameter as a placeholder for when we are given the data to pass into the function, or the argument. Good job including an example!

File: interview-questions.md [hunk 7]
```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Researched answer:
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: .map goes through each value in a array from a function and does not change the original array key and values are the predefined parameters
 
-Researched answer:
+Researched answer: .map is a collections of key value pairs. It does not contain any keys by default. it only contains what is explicitly put into it. It can be any value like functions objecs or any primitive. It iterates in the order of insertion.
```
Comment (Genemichael25) [inline interview-questions.md:28]: See what you can find about .map parameters. Key value pairs are found in objects. .map is a HOF

File: interview-questions.md [hunk 8]
```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+Map creates a new array going through EACH element in a function like dividing all numbers by 2 and logging those answers in its return.
 
 1. What is iteration?
 
-Your answer:
+Your answer:Iteration is process in a function where information or data is repeated and is returned what is requested in that function such as for loops.
 
-Researched answer:
+Researched answer:Iterative refers to a sequence of instructions or code being repeated until a specific end result is achieved. Iterative development is sometimes called circular or evolutionary development. const variable = (function) => {
+                                            for (string) =>{
```
Comment (Genemichael25) [inline interview-questions.md:50]: Researched answers need to be in your own words! We want to know what you understand and not what we can find on the internet and copy/paste.

File: interview-questions.md [hunk 9]
```diff
@@ -8,50 +8,73 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-Your answer:
+Your answer: A method is a const with key and values such as data, arrays, numbers labeled such as
+            const goodFruit = [papaya, apple, orange]
+            A function is a process which takes variables and methods through specific parameters and returning information that is being requested
+
+Researched answer: A method, like a function, is a set of instructions that perform a task. The difference is that a method is associated with an object, while a function is not. 
+var str = 'CodeCADEMY';
+var str1 = str.toLowerCase();
+var str2 = str.toUpperCase();
```
Comment (Genemichael25) [inline interview-questions.md:65]: Same thing here! And if you are going to provide examples, they need to be explained.

**Review by Genemichael25 — CHANGES_REQUESTED** [review]
Dennis I am requesting changes for interview questions regarding iteration and differences between a function and a method. Researched answers need to be in your own words, and should expand on your initial answer. We can use resources to understand the concepts, but copy and paste is not acceptable. Please message me when you have made the changes. I have left a few comments throughout for you to take a look at as well. Currently, this assessment is at a complete level.

**Review by TsunamiCode5 — APPROVED** [review]
Dennis, great job getting through this one! You took the advice and overcame. You have great foundational knowledge and I would like you to build on that! Take some time to read just one of the several different JavaScript methods and get solid on that one, next week another and so on. This assessment comprehension level is a novice but I see you being able to level up within the upcoming assessment. Following the Dreyfus model of skill acquisition here at learn, this is to include pseudo code with proper vocabulary terms, "your answers" written in complete sentences the student's own words that provide partially correct comments, researched answers written in complete sentences in the student's own words that provides additional context, and code that will output correct content in the terminal.

**Pull request merged**